src/on_message.py

```diff
@@ -91,6 +96,60 @@ async def handle_rep(message):
                         member = message.guild.get_member(member)
                         await member.add_roles(role)
 
+def typing_timer(channel_id, user_id, ttl):
+    time.sleep(ttl)
+    if str(channel_id) not in channels_typing:
+        return
+    for x in channels_typing[str(channel_id)]:
+        if x["user_id"] == user_id:
+            channels_typing[str(channel_id)].remove(x)
+            break
+
+@bot.event
+async def on_raw_typing(payload):
+    if (payload.channel_id not in AUTO_SLOWMODE_CHANNELS) or not payload.guild_id:
+        return
+    if str(payload.channel_id) not in channels_typing:
+        channels_typing[str(payload.channel_id)] = []
+    if payload.user_id not in [x["user_id"] for x in channels_typing[str(payload.channel_id)]]:
+        channels_typing[str(payload.channel_id)].append({"user_id": payload.user_id, "startedAt": payload.when})
+    else:
+        for x in channels_typing[str(payload.channel_id)]:
+            if x["user_id"] == payload.user_id:
+                x["startedAt"] = payload.when
+
+    # thread to automatically remove user from typing list after 10 seconds
+    threading.Thread(target=typing_timer, args=(payload.channel_id, payload.user_id, 10)).start()
+
+@tasks.loop(seconds=15)
+async def handle_slowmode():
+    global channels_typing
+    for channel_id in channels_typing:
+        channel = bot.get_channel(int(channel_id))
+        if not channel:
+            return
+        number_of_users_typing = len(channels_typing[channel_id])
+        slowmode = 0
+            
+        match number_of_users_typing:
+            case _ if number_of_users_typing >= 2:
+                slowmode = 3
+            case _ if number_of_users_typing >= 4:
+                slowmode = 5
+            case _ if number_of_users_typing >= 6:
+                slowmode = 10
+            case _ if number_of_users_typing >= 9:
+                slowmode = 15
+            case _ if number_of_users_typing >= 15:
```

Comment:
just use `if` and `elif` statements like this:
```py
if number_of_users_typing < 3: # 3 because 2 users should be allowed to talk without slowmode
    slowmode = 3
elif number_of_users_typing < 5:
    slowmode = 5
# ... more statements
else:
    slowmode = 0